mlflow/data/pandas_dataset.py

```diff
@@ -206,26 +53,16 @@ def from_pandas(
         )
         dataset = mlflow.data.from_pandas(x, targets="Label", predictions="ModelOutput")
     """
-
-    from mlflow.data.code_dataset_source import CodeDatasetSource
-    from mlflow.data.dataset_source_registry import resolve_dataset_source
-    from mlflow.tracking.context import registry
-
-    if source is not None:
-        if isinstance(source, DatasetSource):
-            resolved_source = source
-        else:
-            resolved_source = resolve_dataset_source(
-                source,
-            )
-    else:
-        context_tags = registry.resolve_tags()
-        resolved_source = CodeDatasetSource(tags=context_tags)
-    return PandasDataset(
+    if not isinstance(df, pd.DataFrame):
+        raise TypeError(
+            f"The specified dataframe must be an instance of pandas.DataFrame. Got: {type(df)}."
+        )
+    return from_dataframe(
```

Comment:
To be fair, this would enable also `cuDF` and `modin` dataframes. As this was never discussed, I did not allow for them here